core/src/main/java/io/parsingdata/metal/token/Sub.java

```diff
@@ -32,6 +32,20 @@
 import io.parsingdata.metal.expression.value.OptionalValue;
 import io.parsingdata.metal.expression.value.ValueExpression;
 
+/**
+ * A {@link Token} that specifies a token to be parsed at a specific location
+ * in the input.
```

Comment:
`the input` could also be something like `the current source`, or something similar.